### PR TITLE
ReimbursementPool on Lynx

### PR DESCRIPTION
**Pull request by cygnusv**
**Type of PR:**
- [ ] Bugfix
- [X] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [ ] 2
- [X] 3

**What this does:**
* Introduces ReimbursementPool as a contract in our repo
* Adapts this contract to our contract suite and OpenZeppelin v5
* Deployment scripts and parameters for Lynx testnet
* Script to set `ReimbursementPool` on Coordinator

**Issues fixed/closed:**
* #219 

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Comment by cygnusv** [comment]
Contract deployed in Amoy: https://amoy.polygonscan.com/address/0x53fe19093be1ac0136d33d91e01aba4aa381b87b

**Review by KPrasch — APPROVED** [review]

File: contracts/contracts/coordination/ReimbursementPool.sol [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        isAuthorized[_contract] = true;
+
+        emit AuthorizedContract(_contract);
+    }
+
+    /// @notice Unauthorize a contract that was previously authorized to interact
+    ///         with this reimbursment pool. Can be unauthorized by the
+    ///         owner only.
+    /// @param _contract Authorized contract.
+    function unauthorize(address _contract) external onlyOwner {
```
Comment (derekpierre) [inline contracts/contracts/coordination/ReimbursementPool.sol:89]: Bit of a nitpick. I know this was from a contract copied from Keep, but should this be `deauthorize`? We've used the `authorize/deauthorize` parlance in other areas.

Same comment for the event.

Comment (cygnusv) [reply contracts/contracts/coordination/ReimbursementPool.sol:89]: I agree that there's a divergence wrt to staking contracts but this contract is not related to staking, and I'd prefer if we limit changes unless we really need them.

Comment (derekpierre) [reply contracts/contracts/coordination/ReimbursementPool.sol:89]: Definitely not a "need" 😄

**Review by derekpierre — APPROVED** [review]
🎸

**Review by vzotova — APPROVED** [review]

**Pull request merged**